registry/storage/blobwriter_resumable.go

```diff
@@ -24,12 +21,18 @@ func (bw *blobWriter) resumeDigest(ctx context.Context) error {
 		return errResumableDigestNotAvailable
 	}
 
-	h, ok := bw.digester.Hash().(resumable.Hash)
+	h, ok := bw.digester.Hash().(encoding.BinaryMarshaler)
 	if !ok {
 		return errResumableDigestNotAvailable
 	}
+
+	state, err := h.MarshalBinary()
+	if err != nil {
+		return err
+	}
+
 	offset := bw.fileWriter.Size()
-	if offset == h.Len() {
+	if offset == int64(len(state)) {
```

Comment:
I don't think this is right. In a resumable hash then `Len()` is the number of bytes that have been written to the hash so far. I believe this is just giving you the len of the digest itself.

Reply:
No this is the length of the state not the length of the digest.

Reply:
This is wrong though since the state length is not the offset we are looking for. We may need to implement a digester to track the offset written to the hash